release process: Simplify announcement process (#18614)
coq/coq: dev/doc/release-process.md

- [ ] Push the new tag with `git push upstream VX.X... --dry-run` (remove the `--dry-run` and redo if everything looks OK).
- [ ] Set `is_a_released_version` to `false` (if you forget about it, you'll be reminded by the test-suite failing whenever you try to backport a PR with a changelog entry).
- [ ] Close the milestone.

> - [ ] Send an e-mail on coqdev + coq-club + the Coq announcement category on Discourse (coqdev@inria.fr + coq-club@inria.fr + coq+announcements@discoursemail.com) announcing that the tag has been set so that package managers can start preparing package updates (including a `coq-bignums` compatible version).
>   - for the RC: announce that the release candidate is stable: library authors can safely start preparing compatible releases;

Reviewer: ```suggestion
- [ ] Send an e-mail on coqdev + coq-club + the Coq announcement category on Discourse (coqdev@inria.fr + coq-club@inria.fr + coq+announcements@discoursemail.com) announcing that the tag has been set so that package managers can start preparing package updates (including a `coq-bignums` compatible version), after which library authors can safely start preparing compatible releases.
```
The "sub-bullet" seems strange to me, I think the following is more accurate.
